frontend/admin/resources/js/components/skill/CreateSkill.tsx

```diff
@@ -0,0 +1,257 @@
+}
+
+export const CreateSkillForm: React.FunctionComponent<CreateSkillFormProps> = ({
+  families,
+  handleCreateSkill,
+}) => {
+  const intl = useIntl();
+  const locale = getLocale(intl);
+  const paths = useAdminRoutes();
+  const methods = useForm<FormValues>();
+  const { handleSubmit } = methods;
+  const sortedFamilies = sortBy(families, (family) => {
+    return family.name?.[locale]?.toLocaleUpperCase();
+  });
+
+  const formValuesToSubmitData = (values: FormValues): CreateSkillInput => ({
+    ...values,
+    keywords: values.keywords.split(",").map((key) => key.trim()),
+    families: {
+      sync: values.families,
+    },
+  });
+
+  const onSubmit: SubmitHandler<FormValues> = async (data: FormValues) => {
+    return handleCreateSkill(formValuesToSubmitData(data))
+      .then(() => {
+        navigate(paths.skillTable());
+        toast.success(
+          intl.formatMessage({
+            defaultMessage: "Skill created successfully!",
+            description:
+              "Message displayed to user after skill is created successfully.",
+          }),
+        );
+      })
+      .catch(() => {
+        toast.error(
+          intl.formatMessage({
+            defaultMessage: "Error: creating skill failed",
+            description:
+              "Message displayed to user after skill fails to get created.",
+          }),
+        );
+      });
+  };
+
+  const skillFamilyOptions: Option<string>[] = sortedFamilies.map(
+    ({ id, name }) => ({
+      value: id,
+      label: name?.[locale] || "",
+    }),
+  );
+
+  return (
+    <section>
+      <h2 data-h2-text-align="b(center)" data-h2-margin="b(top, none)">
+        {intl.formatMessage({
+          defaultMessage: "Create Skill",
+          description: "Title displayed on the create a skill form.",
+        })}
+      </h2>
+      <div data-h2-container="b(center, s)">
+        <FormProvider {...methods}>
+          <form onSubmit={handleSubmit(onSubmit)}>
+            <Input
+              id="key"
+              name="key"
+              label={intl.formatMessage({
+                defaultMessage: "Key",
+                description:
+                  "Label displayed on the create a skill form key field.",
+              })}
+              context={intl.formatMessage({
+                defaultMessage:
+                  "The 'key' is a string that uniquely identifies a skill. It should be based on the skills English name, and it should be concise. A good example would be \"information_management\". It may be used in the code to refer to this particular skill, so it cannot be changed later.",
+                description:
+                  "Additional context describing the purpose of the skills 'key' field.",
+              })}
+              type="text"
+              rules={{
+                required: intl.formatMessage(errorMessages.required),
+                pattern: {
+                  value: /^[a-z]+(_[a-z]+)*$/,
```

Comment:
A PR to centralize the regular expressions was just merged in.  Could you please check #1956 instead of using the literal here?